pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -327,7 +327,8 @@ class ClassTypeHintMatch:
                 0: "HomeAssistant",
                 1: "ConfigType",
             },
-            return_type=["DeviceScanner", "DeviceScanner | None"],
+            return_type=["DeviceScanner", None],
+            check_return_type_inheritance=True,
```

Comment:
Since #77606 was merged, we can use the inheritance check

Reply:
That change is definitely fine!